diff --git a/source/tools/toolutil/package.h b/source/tools/toolutil/package.h
index ea60c13a7..38674fcef 100644
--- a/source/tools/toolutil/package.h
+++ b/source/tools/toolutil/package.h
@@ -27,7 +27,7 @@
 
 // .dat package file representation ---------------------------------------- ***
 
-#define STRING_STORE_SIZE 100000
+#define STRING_STORE_SIZE 300000
 #define MAX_PKG_NAME_LENGTH 64
 
 typedef void CheckDependency(void *context, const char *itemName, const char *targetName);
