diff --git a/source/common/uposixdefs.h b/source/common/uposixdefs.h
index 8caf111d8..a7e5e8ddf 100644
--- a/source/common/uposixdefs.h
+++ b/source/common/uposixdefs.h
@@ -48,7 +48,7 @@
 #endif
 
 /*
- * Make sure things like realpath and such functions work.
+ * Make sure things like readlink and such functions work.
  * Poorly upgraded Solaris machines can't have this defined.
  * Cleanly installed Solaris can use this #define.
  *
